tests/acceptance/test_full_configuration.py

```diff
@@ -225,3 +225,57 @@ def test_logprep_exposes_prometheus_metrics(tmp_path):
         r"logprep_tracking_interval_in_seconds.*config_version.*logprep_version.* 1\.0", metrics
     )
     proc.kill()
+
+
+def test_pipeline_with_inline_rule_configs(tmp_path):
+    pipeline_definition = """---
+version: 1
+process_count: 1
+timeout: 0.1
+logger:
+  level: DEBUG
+
+pipeline:
+    - calculator:
+        type: calculator
+        generic_rules:
+            - filter: "message"
+              calculator:
+                target_field: target
+                calc: "1 + 1"
+        specific_rules:
+            - filter: "message1"
+              calculator:
+                target_field: target
+                calc: "1 + 3"
+            - filter: "message1"
+              calculator:
+                target_field: target1
+                calc: "6 + 3" 
+              description: "test description"
+input:
+  kafka:
+    type: confluentkafka_input
+    bootstrapservers:
+      - 127.0.0.1:9092
+    topic: consumer
+    group: cgroup3
+    auto_commit: true
+    session_timeout: 6000
+    offset_reset_policy: smallest
+
+output:
+  opensearch:
+    type: opensearch_output
+    hosts:
+      - 127.0.0.1:9200
+    default_index: processed
+    error_index: errors
+    message_backlog_size: 10000
+    timeout: 10000
+    flush_timeout: 10
+    max_retries: 3
+    user: admin
+    secret: admin
+
+    """
```

Comment:
I think the actual test is missing here :wink:

Reply:
yes you are right. I tested the factory and the functionality is tested in the acceptance tests by changing the quickstart `pipeline.yml`